### ISSUE TITLE
Question: the relationship between docker::image and docker::run

### ISSUE THREAD

**Issue by f4rx** [puppetlabs/puppetlabs-docker#479]
I have manifest:

```puppet
  docker::image { 'f3ex/some-content-nginx':
    ensure => 'latest',
    image_tag => '1',
  } ~>
  docker::run { 'my-nginx':
    image            => 'f3ex/some-content-nginx:1',
  }
```

Expectation: to update/restart a running container on getting a new image.
In the currently: restarting a container on every puppet agent execution.

```bash
Notice: /Stage[main]/Prometheus_node::Alertmanager/Docker::Image[f3ex/some-content-nginx]/Notify[Check if image f3ex/some-content-nginx:1 is in-sync]/message: defined 'message' as 'Check if image f3ex/some-content-nginx:1 is in-sync'
Info: /Stage[main]/Prometheus_node::Alertmanager/Docker::Image[f3ex/some-content-nginx]/Notify[Check if image f3ex/some-content-nginx:1 is in-sync]: Scheduling refresh of Exec[echo 'Update of f3ex/some-content-nginx:1 complete']
Notice: /Stage[main]/Prometheus_node::Alertmanager/Docker::Image[f3ex/some-content-nginx]/Exec[echo 'Update of f3ex/some-content-nginx:1 complete']: Triggered 'refresh' from 1 event
Info: Docker::Image[f3ex/some-content-nginx]: Scheduling refresh of Docker::Run[my-nginx]
Info: Docker::Run[my-nginx]: Scheduling refresh of Service[docker-my-nginx]
Info: Docker::Run[my-nginx]: Scheduling refresh of Exec[docker-my-nginx-systemd-reload]
Notice: /Stage[main]/Prometheus_node::Alertmanager/Docker::Run[my-nginx]/Exec[docker-my-nginx-systemd-reload]: Triggered 'refresh' from 1 event
Notice: /Stage[main]/Prometheus_node::Alertmanager/Docker::Run[my-nginx]/Service[docker-my-nginx]: Triggered 'refresh' from 1 event
```
Cause - the 'exec', that running on every execution: 
```
    notify { "Check if image ${image_arg} is in-sync":
      noop      => false,
    }
    ~>
    exec { "echo 'Update of ${image_arg} complete'":
```
The changes were made in this issue https://github.com/puppetlabs/puppetlabs-docker/issues/316

**Comment by f4rx** [comment]
To prevent starting 'Exec' and downloading new image with --noop, maybe use something like this:
```puppet
    if ! $::clientnoop {
      exec { "echo 'Update of ${image_arg} complete'":
...
```

**Comment by pegasd** [comment]
This is a very inconvenient behavior for us. Even if this gets fixed using the solution suggested by @f4rx , this still throws notifies on every Puppet run (and also returns `2` with `--detailed-exitcodes`).

I believe the issue that was fixed in #297 is a lot more minor that the current behavior.

We're stuck with using `1.1.0` because of this. The way images are updated there seems to me a LOT cleaner.

**Comment by adrianiurca** [comment]
work-related to this issue will be included in PR #705
